### DISCUSSION TITLE
Any reason the `OffsetDateTime::unix_timestamp()` doesn't count `offset`?

### DISCUSSION THREAD

**Discussion by Folyd** [time-rs/time#393] [Question] [answered]
I still get a `UTC` unix timestamp even though I change my offset to `+8`.
```rust
// My local time: Fri Nov 12 2021 17:39:36 GMT+0800 (China Standard Time)
let t = OffsetDateTime::now_utc().replace_offset(UtcOffset::from_hms(8, 0, 0).unwrap())
t.unix_timestamp(); // 1636681176 

// JS
// new Date(1636681176000)
// Fri Nov 12 2021 09:39:36 GMT+0800 (China Standard Time)
```
```rust
pub const fn unix_timestamp(self) -> i64 {
    let days =
        (self.utc_datetime.to_julian_day() as i64 - UNIX_EPOCH_JULIAN_DAY as i64) * 86_400;
    let hours = self.utc_datetime.hour() as i64 * 3_600;
    let minutes = self.utc_datetime.minute() as i64 * 60;
    let seconds = self.utc_datetime.second() as i64;
    days + hours + minutes + seconds
}
```

**Comment by jhpratt** [comment, marked as answer]
The Unix timestamp is by definition UTC and does not need to take the UTC offset into account.

**Reply by Folyd** [reply]
Ok, I see. How can I get the UNIX timestamp in my local timezone?

**Reply by jhpratt** [reply]
Again, the Unix timestamp is by definition UTC. What you're asking fundamentally does not make sense. The Unix timestamp doesn't care about time zones.

**Reply by Folyd** [reply]
Thanks. 👍

**Reply by Folyd** [reply]
Change to `OffsetDateTime::now_utc().to_offset(UtcOffset::from_hms(8, 0, 0).unwrap())` resolve my issue. 😄

**Reply by jhpratt** [reply]
Ah, I was focusing more on the question than the code. If all you're doing is getting the Unix timestamp, though, you don't even need to change the offset. Just call `OffsetDateTime::now_utc().unix_timestamp()`.

**Reply by Folyd** [reply]
Got it. Thanks so much. ❤️
